xdsl/dialects/builtin.py

```diff
@@ -1383,7 +1382,7 @@ def from_list(
         if isinstance(data_type, IntegerType):
             _data = cast(Sequence[int] | Sequence[IntAttr], data)
             return DenseArrayBase.create_dense_int(data_type, _data)
-        elif isattr(data_type, AnyFloatConstr):
+        elif AnyFloatConstr.matches(data_type):
```

Comment:
This one could be an isinstance

Reply:
Argument of type "FloatAttr[Unknown]" cannot be assigned to parameter "data_type" of type "AnyFloat" in function "create_dense_float"

Reply:
Works fine on my computer (maybe you did `isinstance(data_type, FloatAttr)`?)